python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -568,6 +570,31 @@ def agg(self, func):
 
         if self._sort:
             result = result.sort_index()
+        else:
+            if cudf.get_option(
+                "mode.pandas_compatible"
+            ) and not libgroupby._is_all_scan_aggregate(normalized_aggs):
+                # Even with `sort=False`, pandas clearly documents and
+                # ensures that the ordering inside the group by columns
+                # is preserved(i.e., it is not non-deterministic).
+                if not isinstance(result_index, cudf.MultiIndex):
+                    lcol, rcol = _match_join_keys(
+                        self.grouping.keys.drop_duplicates()._column,
+                        result_index._column,
+                        "inner",
+                    )
+                    _, indices = libcudf.join.join([lcol], [rcol], how="inner")
+                else:
+                    left_cols = list(
+                        self.grouping.keys.drop_duplicates()._data.columns
+                    )
+                    right_cols = list(result_index._data.columns)
+                    _, indices = libcudf.join.join(
+                        left_cols, right_cols, how="inner"
+                    )
+
+                new_index = result_index[indices]
+                result = result.reindex(new_index)
```

Comment:
Rather than `reindex`, are we not able to reorder `result` using `indices`? Or does that not do the right thing?

Reply:
We can directly use `take`, changed it.